src/test/jtx/impl/Env.cpp

```diff
@@ -298,28 +320,27 @@ Env::submit(JTx const& jt)
             jt.stx->add(s);
             auto const jr = rpc("submit", strHex(s.slice()));
 
-            std::tie(ter_, didApply) = parseResult(jr);
+            parsedResult = parseResult(jr);
+            ter_ = parsedResult.ter;
 
             return jr;
         }
         else
         {
             // Parsing failed or the JTx is
             // otherwise missing the stx field.
-            ter_ = temMALFORMED;
-            didApply = false;
+            parsedResult.ter = ter_ = temMALFORMED;
```

Comment:
No, because this is a different situation. This check is done _before_ any RPC calls are made, so `RPC_FAILED` would be inappropriate. Additionally, the only way this should happen is if the `JTx` object is unabled to be parsed into an `STTx`, which would mean that it's `MALFORMED`.